instrumentation-genai/opentelemetry-instrumentation-vertexai-v2/src/opentelemetry/instrumentation/vertexai_v2/__init__.py

```diff
@@ -0,0 +1,450 @@
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_async",
+        "span_name": "vertexai.predict_async",
+        "is_async": True,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_streaming",
+        "span_name": "vertexai.predict_streaming",
+        "is_async": False,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_streaming_async",
+        "span_name": "vertexai.predict_streaming_async",
+        "is_async": True,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "ChatSession",
+        "method": "send_message",
+        "span_name": "vertexai.send_message",
+        "is_async": False,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "ChatSession",
+        "method": "send_message_streaming",
+        "span_name": "vertexai.send_message_streaming",
+        "is_async": False,
+    },
+]
+
+
+def should_send_prompts():
+    # Previously was opt-in by the following check for privacy reasons:
+    #
+    # return (
+    #     os.getenv("TRACELOOP_TRACE_CONTENT") or "true"
+    # ).lower() == "true" or context_api.get_value(
+    #     "override_enable_content_tracing"
+    # )
+    return True
+
+
+def is_streaming_response(response):
+    return isinstance(response, types.GeneratorType)
+
+
+def is_async_streaming_response(response):
+    return isinstance(response, types.AsyncGeneratorType)
+
+
+def _set_span_attribute(span, name, value):
+    if value is not None:
+        if value != "":
+            span.set_attribute(name, value)
+
+
+def _set_input_attributes(
+    span, event_logger: EventLogger, args, kwargs, llm_model
+):
+    if should_send_prompts() and args is not None and len(args) > 0:
+        prompt = ""
+        for arg in args:
+            if isinstance(arg, str):
+                prompt = f"{prompt}{arg}\n"
+            elif isinstance(arg, list):
+                for subarg in arg:
+                    prompt = f"{prompt}{subarg}\n"
+
+        # _set_span_attribute(
+        #     span,
+        #     f"{SpanAttributes.LLM_PROMPTS}.0.user",
+        #     prompt,
+        # )
+        if prompt:
+            event_logger.emit(
+                user_event(
+                    gen_ai_system=gen_ai_attributes.GenAiSystemValues.VERTEX_AI.value,
+                    content=prompt,
+                    span_context=span.get_span_context(),
+                )
+            )
+
+        # Copied from openllmetry logic
+        # https://github.com/traceloop/openllmetry/blob/v0.33.12/packages/opentelemetry-instrumentation-vertexai/opentelemetry/instrumentation/vertexai/__init__.py#L141-L143
+        # I guess prompt may be in kwargs instead or in addition?
+        prompt = kwargs.get("prompt")
+        if prompt:
```

Comment:
why did we handle `prompt` twice here? Another one is in Line 165

Reply:
I added a comment, was wondering the same thing lol hoping to get clarity from TraceLoop folks who originally wrote this code.

Reply:
@nirga ^^

Reply:
@aabmass @gyliu513 it's to support the original language_models API:
https://cloud.google.com/vertex-ai/generative-ai/docs/reference/python/latest/vertexai.language_models.TextGenerationModel#vertexai_language_models_TextGenerationModel_predict